`azurerm_storage_account` add block `azure_files_identity_based_authentication` and block `routing_preference` (#8961)
terraform-providers/terraform-provider-azurerm: website/docs/r/storage_account.html.markdown

---

A `azure_files_identity_based_authentication` block supports the following:

> * `directory_service_options` - (Required) Specifies the directory service used. Possible values are `DirectoryServiceOptionsAADDS` and `DirectoryServiceOptionsAD`.

Reviewer: ```suggestion
* `directory_service_options` - (Required) Specifies the directory service used. Possible values are `AADDS` and `AD`.
```

> * `active_directory` - (Optional) A `active_directory` block as defined below. Required when `directory_service_options` is `DirectoryServiceOptionsAD`.

Reviewer: ```suggestion
* `active_directory` - (Optional) A `active_directory` block as defined below. Required when `directory_service_options` is `AD`.
```

---

A `active_directory` block supports the following:

* `azure_storage_sid` - (Required) Specifies the security identifier (SID) for Azure Storage.

* `domain_name` - (Required) Specifies the primary domain that the AD DNS server is authoritative for.

* `domain_sid` - (Required) Specifies the security identifier (SID).

* `domain_guid` - (Required) Specifies the domain GUID.

> * `forest_name` - (Required) Specifies the Active Directory forest to get.

Reviewer: ```suggestion
* `forest_name` - (Required) Specifies the Active Directory forest.
```

* `net_bios_domain_name` - (Required) Specifies the NetBIOS domain name.

---

A `routing_preference` block supports the following:

> * `publish_internet_endpoints` - (Optional) Are internet routing storage endpoints to be published? Defaults to `false`.

Reviewer: ```suggestion
* `publish_internet_endpoints` - (Optional) Should internet routing storage endpoints be published? Defaults to `false`.
```

> * `publish_microsoft_endpoints` - (Optional) Are microsoft routing storage endpoints  to be published? Defaults to `false`.

Reviewer: ```suggestion
* `publish_microsoft_endpoints` - (Optional) Should Microsoft routing storage endpoints be published? Defaults to `false`.
```
